index.d.ts

```diff
@@ -9,68 +12,135 @@ interface BackendOptions {
+  parsePayload?(
+    namespace: string,
+    key: string,
+    fallbackValue?: string
+  ): RequestPayload;
+
   /**
    * define a custom xhr function
    * can be used to support XDomainRequest in IE 8 and 9
    */
   request?(
     options: BackendOptions,
     url: string,
-    payload: {} | string,
-    callback: RequestCallback,
+    payload: RequestPayload | undefined,
+    callback: RequestCallback
   ): void;
+
+  /**
+   * periodically reload the localisation files:
+   */
+  reloadInterval: false | number;
+
+  /**
+   * set custom request headers
+   * overrides headers previously set if the key is a duplicate of a header set internally:
+   */
+  customHeaders?: { [key: string]: string };
+
   /**
    * adds parameters to resource URL. 'example.com' -> 'example.com?v=1.3.5'
    */
   queryStringParams?: { [key: string]: string };
 
-  customHeaders?: { [key: string]: string };
+  /**
+   * allow cross domain requests, only used for non-fetch APIs. See requestOptions when using fetch().
+   */
+  crossDomain?: boolean;
+
+  /**
+   * allow credentials on cross domain requests, only used for non-fetch APIs. See requestOptions when using fetch().
+   */
+  withCredentials?: boolean;
+
+  /**
+   * forces mime-type to be 'application/json', only used for non-fetch APIs
+   */
+  overrideMimeType?: boolean;
+
+  /**
+   * Set custom request options when using fetch() for loading localisation files.
+   */
+  requestOptions?: RequestInit | ((payload: RequestPayload) => RequestInit);
 }
 
+type RequestPayload = { [key: string]: any };
 type RequestCallback = (error: any, response: RequestResponse) => void;
 
 interface RequestResponse {
-  status: number,
-  data: string
+  status: number;
+  // optional, only present if the request was successful:
+  data?: string;
 }
 
-export default class I18NextHttpBackend implements BackendModule<BackendOptions> {
+export default class I18NextHttpBackend
+  implements BackendModule<BackendOptions> {
   constructor(services?: any, options?: BackendOptions);
   init(services?: any, options?: BackendOptions): void;
-  readMulti(languages: string[], namespaces: string[], callback: ReadCallback): void;
+  readMulti(
+    languages: string[],
+    namespaces: string[],
+    callback: ReadCallback
+  ): void;
   read(language: string, namespace: string, callback: ReadCallback): void;
-  loadUrl(url: string, callback: ReadCallback, languages?: string | string[], namespaces?: string | string[]): void;
-  create(languages: string | string[], namespace: string, key: string, fallbackValue: string): void;
-  type: 'backend';
+  loadUrl(
+    url: string,
+    callback: ReadCallback,
+    languages?: string | string[],
+    namespaces?: string | string[]
+  ): void;
+  create(
+    languages: string | string[],
+    namespace: string,
+    key: string,
+    fallbackValue: string
+  ): void;
+  type: "backend";
   services: any;
   options: BackendOptions;
 }
+
+declare module "i18next" {
```

Comment:
In theory, this should make i18next's `init` method correctly typed with the options defined here, but for some reason, this isn't working. I have tweeted looking for help: https://twitter.com/thisismissem/status/1303470827627835394?s=21

Reply:
The goal of this is to have these types argument the types of the i18next module, so that it can have correct types for the `backend` keys in the init options.